### DISCUSSION TITLE
How can I get a list of all columns in a sheet?

### DISCUSSION THREAD

**Discussion by halloleo** [saulpw/visidata#1081] [Q&A] [answered]
Hi there

I have quite a few sheets with **lots of** columns, and to make it worse these columns' names are rather long. So it is hard to get an overview over all columns.

Is there a way to see a (vertical) list of all columns? Something similar to what `csvcut -n <file.csv>` outputs?

Thanks for any pointers.

**Comment by ajkerrigan** [comment, marked as answer]
Hi there @halloleo - there are a couple good ways to get column overviews in VisiData. Using some [sample data](https://github.com/saulpw/visidata/blob/stable/sample_data/benchmark.csv) as an example, your `csvcut -n` reference shows:

```
> csvcut -n sample_data/benchmark.csv 
  1: Date
  2: Customer
  3: SKU
  4: Item
  5: Quantity
  6: Unit
  7: Paid
```

With that file open in VisiData, `Shift+C` opens a "Columns" sheet that lists column names vertically along with a number of other column-level properties.

```
 name     ║ width#| height#| type | fmtstr | value                                | expr | aggregators ║
 Date     ║    18 |      1 |      |        | 7/3/2018 1:47p                       | 0   #|             ║
 Customer ║    20 |      1 |      |        | Robert Armstrong                     | 1   #|             ║
 SKU      ║     9 |      1 |      |        | FOOD213                              | 2   #|             ║
 Item     ║    20 |      1 |      |        | BFF Oh My Gravy! Beef & Salmon 2.8oz | 3   #|             ║
 Quantity ║    10 |      1 |      |        | 4                                    | 4   #|             ║
 Unit     ║    11 |      1 |      |        | $12.95                               | 5   #|             ║
 Paid     ║    11 |      1 |      |        | $51.8                                | 6   #|             ║
```

That's a useful way to see a vertical list of columns. Depending on your motivation when you're looking for a column overview, `Shift+I` can also be handy. That opens a "Describe" sheet, which includes summary information about the _contents_ of each column.

```
 column   ║ errors♯| nulls♯| distinct♯| mode              ~| min    %| max    %| sum               %| median%| mean  %| stdev %
 Date     ║      0 |     0 |       45 | 7/10/2018 5:23p    |        !|        !|                   !|       !|       !|       !║
 Customer ║      0 |     0 |       15 | Robert Armstrong   |        !|        !|                   !|       !|       !|       !║
 SKU      ║      0 |     0 |       32 | FOOD122            |        !|        !|                   !|       !|       !|       !║
 Item     ║      0 |     0 |       38 | Food, Senior Wet C…|        !|        !|                   !|       !|       !|       !║
 Quantity ║      0 |     0 |       10 | 1                  |    1.00 |  144.00 |             401.00 |   2.00 |   8.18 |  21.97 ║
 Unit     ║      0 |     0 |       26 | 12.95              |    0.00 | 2495.99 |           11413.83 |  12.95 | 232.94 | 596.19 ║
 Paid     ║      0 |     0 |       34 | 51.8               | -2300.00| 3602.10 |           10664.00 |  51.80 | 217.63 | 836.31 ║
```

It seems like `Shift+C` and/or `Shift+I` could help give you an overview with names that fit on your screen! :)

**Reply by halloleo** [reply]
Hey, thanks for this! `Shift-C` is _exactly_ what I had in mind. 

I didn't find it in the doco, because I was searching for something like "List of columns", but now I can see that the **Column Sheet** is what I was looking for....

**Reply by ghost** [reply]
You might also find the following commands helpful once you're on the Column Sheet:

`gs` (select all rows)
`gzY` (copy all selected rows in the current column to the system clipboard) (Note the uppercase `Y`)

**Reply by halloleo** [reply]
Thanks David. Make sense.

(I am quite intrigued by visidata's keyboard shortcuts: They appear hard to be memorized about following an easy-to-grasp structure making it easier than you think...

**Reply by saulpw** [reply]
Thanks, @halloleo...I put a lot of thought into the design of the shortcuts, so it's good to know you find them easier than they seem at first glance!
